### PR TITLE
Add new 3.7 opcode support.

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

This seemed to be the cheapest way to enable the new op codes,
improvements are welcomed!

### CODE REVIEW

**Comment by stonebig** [comment]
I can't wait to try an alpha numba python-3.8 wheel.

**Comment by stuartarchibald** [comment]
Thanks for the review @seibert. The 3.7 build isn't building with 3.7 yet as there's no llvmlite for 3.7 in the Numba channel, as a result there's a giant downgrade of packages to 3.6 half way through the build. I'll try adding a 3.7 build to the Numba farm to ensure these packages are created and used, we shouldn't merge until this is done.

**Comment by stuartarchibald** [comment]
I pushed this through the Numba build farm. The 32bit OS builders failed as there's no shipped Py 3.7 packages available on the `c3i_test` channel yet, however the rest of the builds went fine.

**Comment by sklam** [comment]
I think we should leave a note about how we are handling the new opcodes differently.  The `LOAD_METHOD` is different from the `LOAD_ATTR` that it is pushing two values (`self` and the unbound method) onto the stack.  It allows `CALL_METHOD` to call unbound method directly.  

Beside that, we should ensure that the new opcodes are exercised in one of the objectmode test.

**Comment by stuartarchibald** [comment]
The test `numba.tests.test_object_mode.TestObjectModeInvalidRewrite.test_static_setitem_in_objmode_fallback` uses both `LOAD_METHOD` and `CALL_METHOD` op codes under Python 3.7. It fails to run without this patch, and just adding `LOAD_METHOD` support causes subsequent failure on `CALL_METHOD`.

**Comment by stuartarchibald** [comment]
@sklam do you have a view as to where you'd like this note to go? In the `interpreter.py` or `dataflow.py` source perhaps?

**Comment by stuartarchibald** [comment]
Thanks @sklam (and for out of band discussion on the motivation). https://github.com/numba/numba/pull/2957/commits/c3b4d762cacc21c375c1c88f223fa58dd0f6339b should complete 3.7 support (assuming no surprises from 32bit builds).

File: numba/interpreter.py [hunk 1]
```diff
@@ -985,3 +985,18 @@ def op_LIST_APPEND(self, inst, target, value, appendvar, res):
...
         appendinst = ir.Expr.call(self.get(appendvar), (value,), (), loc=self.loc)
         self.store(value=appendinst, name=res)
+
+
+    # NOTE: The LOAD_METHOD opcode is implemented as a LOAD_ATTR for ease,
+    # however this means a new object (the bound-method instance) could be
+    # created. Conversely, using a pure LOAD_METHOD no intermediary is present
+    # and it is essentially like a pointer grab and forward to CALL_METHOD. The
+    # net outcome is that the implementation in Numba produces the same result,
+    # but in object mode it may be that it runs mode slowly than it would if
```
Comment (seibert) [inline numba/interpreter.py]: s/mode/more/

**Comment by seibert** [comment]
one typo nit, and then this is ready to merge!

**Pull request merged**